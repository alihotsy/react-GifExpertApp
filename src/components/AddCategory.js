import React, {useState} from 'react'
import PropTypes from 'prop-types';

const useMyInput = (setCategories) =>{
    const [inputValue,setInputValue] = useState('');
    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const handleInput = (e) =>{
        if(e.key === "Enter"){
            if(e.target.className === "input1"){
                if(inputValue.trim().length > 2){
                    setCategories(cats => [inputValue,...cats]);
                    setInputValue('');
                }
            } else{
                setCategories(cats => cats.filter(n => n !== inputValue))
                setInputValue('');
            }
           
        }
    }
    const handleSubmit = (e) =>{
        e.preventDefault();
        
    }
    return {
        handleInputChange,
        handleSubmit,
        inputValue,
        handleInput
    }
}
export const AddCategory = ({setCategories}) => {
    const {handleSubmit} = useMyInput(setCategories)
    const input1 = useMyInput(setCategories);
    const input2 = useMyInput(setCategories);
    return (
        <form onSubmit = {handleSubmit}> 
            <p>{input1.inputValue}</p>
            <label>Ingresa una serie:</label>
            <input
             className = "input1"
             type="text"
             value = {input1.inputValue}
             onChange = {input1.handleInputChange}
             onKeyPress = {input1.handleInput}
            />
            <label>Borra una serie de la lista:</label>
            <input
             type="text2"
             value = {input2.inputValue}
             onChange = {input2.handleInputChange}
             onKeyPress = {input2.handleInput}
            />
            <button className="boton">Borrar</button>
        </form>    
        
        
    )
}
AddCategory.propTypes ={
    setCategories: PropTypes.func.isRequired
}