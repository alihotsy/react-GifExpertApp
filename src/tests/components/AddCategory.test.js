import React from 'react';
import { shallow } from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('Pruebas en <AddCategory/>', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories}/>);

    beforeEach(()=>{
       jest.clearAllMocks();
       wrapper = shallow(<AddCategory setCategories={setCategories}/>);
    });
    test('Debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    test('Debe de cambiar la caja de texto o input', () => {
        const input = wrapper.find('input').at(0);
        const value = 'Hola mundo';
        input.simulate('change', {target:{value}});
        expect(wrapper.find('p').text().trim()).toBe(value);
    })
    test('NO debe postear la información con keyPress', () => {
        // wrapper.find('form').simulate('submit', {preventDefault(){}})
        wrapper.find('input').at(0).simulate('keyPress',{key:'Enter',target:'input'});
        expect(setCategories).toHaveBeenCalled();
    })
    test('Debe de llamar el setCategories y limpiar la caja de texto', () => {
        const value = "Hola Mundo";
        const input = wrapper.find('input').at(0);
        //Simular el inputChange
        input.simulate('change',{target:{value}});
        //Simular el KeyPress
        input.simulate('keyPress',{key:'Enter',target:'input'});
        // setCategories se debe de llamar
        expect(setCategories).toHaveBeenCalled();
        // El valor del input debe de estar vacío ''
        expect(input.prop('value')).toBe('');
        // setCategories se debe llamar 1 vez:
        expect(setCategories).toHaveBeenCalledTimes(1);
        //Se espera que se llame una función:
        expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
    })
    
    
})
