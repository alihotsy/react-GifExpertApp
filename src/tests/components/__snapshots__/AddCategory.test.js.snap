// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <AddCategory/> Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <p />
  <label>
    Ingresa una serie:
  </label>
  <input
    className="input1"
    onChange={[Function]}
    onKeyPress={[Function]}
    type="text"
    value=""
  />
  <label>
    Borra una serie de la lista:
  </label>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    type="text2"
    value=""
  />
  <button
    className="boton"
  >
    Borrar
  </button>
</form>
`;
